app/Models/Asset.php

```diff
@@ -313,9 +313,14 @@ public function checkOut($target, $admin = null, $checkout_at = null, $expected_
         }
 
         if ($this->save()) {
+            if (is_integer($admin)){
+                $checkedOutBy = User::findOrFail($admin);
 
-            event(new CheckoutableCheckedOut($this, $target, Auth::user(), $note));
-
+                event(new CheckoutableCheckedOut($this, $target, $checkedOutBy, $note));
+            } else {
```

Comment:
I think one more clause would make this perfect - 

I expect that the `$admin` parameter is supposed to be either an integer ID (as in your case), or an actual `User` object.

I think you can do something where if the `$admin` passed in is `null`, then we initialize it to `Auth::user()`.

And then your same `is_integer()` check could instead set `$admin` to `findOrFail($admin)` just how you're doing it here.

And then there's only *one* event dispatch, (using `$admin` as the parameter).

And it will handle all three cases: null, an integer, or a User object.

Make sense?

Reply:
I make the suggested changes, but I keep the `$checkedOutBy` variable because I think it's more readable, I'm open to change it if doesn't makes sense, of course. Thanks for your time reading and commenting my PRs!!

Reply:
I agree, `$checkedOutBy` does certainly make for a very readable variable. Good call.

Reply:
There's only one more part we've got to handle here, because I think there are *three* cases for the `$admin` variable - 

1. `$admin` is `null` (or not passed at all). You already handle that.
2. `$admin` is an integer. I think you've got that handled too.
3. `$admin` is an `App\Models\User` object. This one I think you're missing.

I was wondering if the third item here is still an actual problem in Snipe-IT, so I ran this:

`grep -r -e '->checkOut' .|grep -v storage/debugbar`

(That's doing a recursive grep for `->checkOut` which may hit for more than just Assets. And then I'm skipping the contents of `storage/debugbar` because it makes my output ugly :) )

```
./app/Importer/AssetImporter.php:                $asset->fresh()->checkOut($target);
./app/Http/Controllers/Api/AssetsController.php:                $asset->checkOut($target, Auth::user(), date('Y-m-d H:i:s'), '', 'Checked out on asset creation', e($request->get('name')));
./app/Http/Controllers/Api/AssetsController.php:                    $asset->checkOut($target, Auth::user(), date('Y-m-d H:i:s'), '', 'Checked out on asset update', e($request->get('name')), $location);
./app/Http/Controllers/Api/AssetsController.php:        if ($asset->checkOut($target, Auth::user(), $checkout_at, $expected_checkin, $note, $asset_name, $asset->location_id)) {
./app/Http/Controllers/Assets/AssetsController.php:                    $asset->checkOut($target, Auth::user(), date('Y-m-d H:i:s'), $request->input('expected_checkin', null), 'Checked out on asset creation', e($request->get('name')), $location);
./app/Http/Controllers/Assets/AssetCheckoutController.php:            if ($asset->checkOut($target, $admin, $checkout_at, $expected_checkin, e($request->get('note')), $request->get('name'))) {
./app/Http/Controllers/Assets/BulkAssetsController.php:                    $error = $asset->checkOut($target, $admin, $checkout_at, $expected_checkin, e($request->get('note')), null);
./app/Services/PredefinedKitCheckoutService.php:                    $error = $asset->checkOut($user, $admin, $checkout_at, $expected_checkin, $note, null);
./tests/unit/AssetTest.php:         $asset->checkOut($target, $adminUser);
./tests/unit/AssetTest.php:         $asset->checkOut($target, $adminUser);
./tests/unit/AssetTest.php:         $target->checkOut($target, $adminUser);
./tests/unit/AssetTest.php:         $asset->checkOut($target, $adminUser);
./tests/unit/NotificationTest.php:        $asset->checkOut($user, 1);
```

There's definitely a couple of times there where we are calling the `checkOut()` method and manually passing in an `$admin` parameter, and that parameter definitely *does* look like a `User` object. So we'll need to handle that too. Since in just about every instance the actual *value* that we're passing through is actually just the value of  `Auth::user()`, I _suspect_ that your code will actually work out OK, but I'd hate to set us up for a bug in the future where we try to pass a User other than the logged-in `Auth::user()` and have things act weird or unexpectedly.

Ping me if you get stuck implementing this last piece.